TUTORIAL.md

```diff
@@ -0,0 +1,245 @@
+
+In order to understand `accepts_nested_attributes_for`, we need to take a look at our models associations and the methods that were made available to us.
+
+***note that we have added `accepts_nested_attributes_for :tags` to our `Post` model**
+
+##Step 4: Set up Post model
+### Models
+
+```ruby
+class Post < ActiveRecord::Base
+  has_many :post_tags
+  has_many :tags, :through => :post_tags
+
+  accepts_nested_attributes_for :tags
+end
+
+class PostTag < ActiveRecord::Base
+  belongs_to :post
+  belongs_to :tag
+end
+
+class Tag < ActiveRecord::Base
+  has_many :post_tags
+  has_many :posts, :through => :post_tags
+end
+```
+By creating these associations, Rails gives us the methods below.
+
+```ruby
+tag_ids()                    Post (Post::GeneratedAssociationMethods)
+tag_ids=(ids)                Post (Post::GeneratedAssociationMethods)
+tags(*args)                  Post (Post::GeneratedAssociationMethods)
+tags=(value)                 Post (Post::GeneratedAssociationMethods)
+tags_attributes=(attributes) Post (Post::GeneratedAssociationMethods)
+```
+What we are interested in right now is `tags_attributes=(attributes)`
+
+As mentioned before, our `fields_for` is looking for `accepts_nested_attributes_for :tags`. But what exactly does this macro give us? Let's define our own and see.
+
+```ruby
+def tags_attributes=(attributes_hash)
+  attributes_hash.values.each do |attributes|
+    self.tags.build(attributes)
+  end
+end
+```
+If we create three new tags and look at params in our console, we will see the following in our `attributes_hash`.
+
+```ruby
+{
+    "0" => {
+        "name" => "tag1"
+    },
+    "1" => {
+        "name" => "tag2"
+    },
+    "2" => {
+        "name" => "tag3"
+    }
+}
+```
+Great! It looks like our `tags_attributes=` method is iterating over our new tags, grabbing the values and building our tags. Then when `@post` is saved, it calls our `tag_attributes=` method in the `Post` model.
+
+
+Let's take a look at what our params would look like if we submit this form.
+
+####Params
+
+```ruby
+{
+                  "utf8" => "✓",
+    "authenticity_token" => "VVPsqwm1XkDsJ1XQjf4vpeLEQdNXl+4hAqvU3gL/RIb5v8RKbcmA7r3gRT6eYM/g9wHY5Ymi6w/PPKuHea1XYg==",
+                  "post" => {
+                   "name" => "Hello World!",
+                "content" => "We are learning about fields_for!",
+        "tags_attributes" => {
+            "0" => {
+                "name" => "tag1"
+            },
+            "1" => {
+                "name" => "tag2"
+            },
+            "2" => {
+                "name" => "tag3"
+            }
+        }
+    },
+                "commit" => "Create Post",
+            "controller" => "posts",
+                "action" => "create"
+}
+```
+As you can see, along with our post `name` and `content` we have a nested hash called `tags_attributes`. These attributes will be written via our `accepts_nested_attributes_for` or custom `tags_attributes=` method.
```

Comment:
How would you have to modify your controller if you were to use the `#tags_attributes` method you wrote? I think that's pretty interesting :)